src/marionette.rs

```diff
@@ -269,7 +273,26 @@ impl MarionetteHandler {
                          capabilities: &NewSessionParameters) -> WebDriverResult<()> {
         let profile = try!(self.load_profile(capabilities));
         let args = try!(self.load_browser_args(capabilities));
-        match self.start_browser(profile, args) {
+        let launcher = if self.connect_existing {
+            BrowserLauncher::None
+        } else {
+            let binary = if let Some(binary_capability) = capabilities.get("firefox_binary") {
+                Some(PathBuf::from(try!(binary_capability
+                                        .as_string()
+                                        .ok_or(WebDriverError::new(
+                                            ErrorStatus::InvalidArgument,
+                                            "'firefox_binary' capability was not a string")))))
+            } else {
+                self.binary.as_ref().map(|x| x.clone())
+            };
+            if let Some(path) = binary {
+                BrowserLauncher::BinaryLauncher(path)
+            } else {
+                return Err(WebDriverError::new(ErrorStatus::UnknownError,
```

Comment:
Could we just try use the default install paths for each OS here instead of erroring?